roofit/roofitcore/src/RooAbsReal.cxx

```diff
@@ -5045,7 +4950,10 @@ RooSpan<double> RooAbsReal::evaluateSpan(BatchHelpers::RunContext& evalData, con
     }
   }
 
-  const auto dataSize = std::max<std::size_t>(1, BatchHelpers::findSmallestBatch(leafValues));
+  size_t dataSize=1;
+  for (auto& i:leafValues) {
+    dataSize=std::max(dataSize, i.size());
+  }
```

Comment:
Trying to make as many helpers `const` as possible, one would write:
```c++
const std::size_t dataSize = std::max_element(leafValues.begin(), leafValues.end(),
    [](const RooSpan<const double>& span){ return span.size(); } )->size();
```
This requires probably `#include <algorithm>`, and I leave it to you to decide if that's readable enough to use it here.